oak_runtime/build.rs

```diff
@@ -37,4 +38,30 @@ fn main() {
             ..Default::default()
         },
     );
+
+    // Copy introspection_browser_client files (generated by the runner) to the
+    // out dir. In case these files don't yet exist, such as when running unit
+    // tests, create placeholder files. This ensures unit tests etc can be run
+    // without needing to build the frontend client assets.
+    const INTROPSECTION_CLIENT_DIR: &str = "src/introspection_browser_client/dist";
+    const INTROPSECTION_CLIENT_FILES: &[&str] = &["index.html", "index.js"];
+    let out_dir = env::var("OUT_DIR").unwrap();
+    fs::create_dir_all(&format!("{}/{}", out_dir, INTROPSECTION_CLIENT_DIR))
+        .expect("unable to create introspection client out directory");
+    for client_file in INTROPSECTION_CLIENT_FILES {
+        let path = format!("{}/{}", INTROPSECTION_CLIENT_DIR, client_file);
+        let out_path = format!("{}/{}", out_dir, path);
+        let mut out_file = fs::OpenOptions::new()
+            .create(true)
+            .write(true)
+            .truncate(true)
+            .open(&out_path)
+            .expect("unable to open/create data file");
+        if let Ok(mut source_file) = fs::File::open(&path) {
+            io::copy(&mut source_file, &mut out_file).expect("failed to copy data after opening");
+        }
+        // Tell cargo to rerun this build script if the client_file has changed.
+        // https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorerun-if-changedpath
+        println!("cargo:rerun-if-changed={}", path);
```

Comment:
This is interesting, in fact I realised I don't actually know how `include_bytes` by itself would handle it; if the included file changed, would cargo notice, without this line?

Reply:
Cargo does notice is the file required by `include_bytes` changed. 

The purpose of this line is a bit different though:

It copies the actual client file in `oak_runtime/src/introspection_browser_client/dist` into the cargo out directory. The `include_bytes` line in `introspect.rs` then reads that file from the cargo out directory and includes it in the binary. 

This build rule therefore takes place before `include_bytes` ever comes into play, and needs to be rerun if the file in `oak_runtime/src/introspection_browser_client/dist` changes.